### PR TITLE
Feature/72 create context service interface

### PR DESCRIPTION
**Pull request by rmakara**
**Changes:**
* New `services/contextService`.
* `getAvailableCurrencies()` method.
* `setCurrentCurrency()` method.
* `getAvailableLanguages()` method.
* `setCurrentLanguage()` method.
* `getAvailableCountries()` method.
* `getAvailablePaymentMethods()` method.
* `setCurrentShippingMethod()` method.
* `getAvailableShippingMethods()` method.
* `setCurrentShippingMethod()` method.
* Tests.

Patryk also extracted contextToken to global config.

**Issue:** 
#72

### CODE REVIEW

File: packages/shopware-6-client/src/services/contextService.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  return resp.data;
+};
+
+interface ContextService {
+  getCurrencies: () => Promise<SearchResult<Currency[]>>;
+  getLanguages: () => Promise<SearchResult<Language[]>>;
+  getCountries: () => Promise<SearchResult<Country[]>>;
+  getPaymentMethods: () => Promise<SearchResult<PaymentMethod[]>>;
+  getShippingMethods: () => Promise<SearchResult<ShippingMethod[]>>;
```
Comment (mkucmus) [inline packages/shopware-6-client/src/services/contextService.ts]:
```suggestion
  getAvailableCurrencies: () => Promise<SearchResult<Currency[]>>;
  getAvailableLanguages: () => Promise<SearchResult<Language[]>>;
  getAvailableCountries: () => Promise<SearchResult<Country[]>>;
  getPaymentMethods: () => Promise<SearchResult<PaymentMethod[]>>;
  getAvailableShippingMethods: () => Promise<SearchResult<ShippingMethod[]>>;
```
having those methods in cotext is totally fine for me, it's directly related, but under the word `context` we should rather have something related to the customer's current session. 
We could have all those methods with the keyword "available" and also with the "current" word when the SW Core Team makes the methods that responses the current context's details so we don't have any at the moment.

File: packages/shopware-6-client/src/services/contextService.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+} from "../endpoints";
+import { Country } from "../interfaces/models/system/country/Country";
+import { ShippingMethod } from "../interfaces/models/checkout/shipping/ShippingMethod";
+import { PaymentMethod } from "../interfaces/models/checkout/payment/PaymentMethod";
+import { Language } from "../interfaces/models/framework/language/Language";
+import { SearchResult } from "../interfaces/response/SearchResult";
+
+const getCurrencies = async function(): Promise<SearchResult<Currency[]>> {
+  const resp = await apiService.get(
+    config.endpoint + getContextCurrencyEndpoint()
```
Comment (mkucmus) [inline packages/shopware-6-client/src/services/contextService.ts]: please use template strings like this
```suggestion
    ${config.endpoint}${getContextCurrencyEndpoint()}
```

Comment (rmakara) [reply packages/shopware-6-client/src/services/contextService.ts]: Done :)

**Comment by rmakara** [comment]
**Changes:**
* Adjusted the code and tests just like in Categories and Products.
* I am not sure about splitting the tests of `Set` methods into 2 described. The first one with contextToken given and the second without contextToken given - as we use these setters as logged and not-logged user.

**Pull request merged**